### DISCUSSION TITLE
Exectute package on Windows

### DISCUSSION THREAD

**Discussion by stilobique** [photoview/photoview#328] [General]
Hello,
i'm package your apps to use-it on a Windows machine. [With an previous issue it's solve](https://github.com/photoview/photoview/issues/70), i can run a compilation and launch a server. 

However, when my package is launch ; Firefox get this issue

```
CreateFile \ui\C:: La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte.
```
I think it's a simple path error. But i don't understand why exactly. Wrong `.env` setup ? The database use an SQLite way, and generate it without error.

The windows terminal give no error. Maybe it's possible to create a log, i don't find a way to setup it.

Thanks for your work, and thanks to read me :) .

**Comment by Darthagnon** [comment]
I've had the same problem, trying to run windows builds.

I suspect that something is trying to make folders with a colon in the foldername, e.g. ```C:```, which is not allowed on Windows.

In my [earlier experimentation with Windows builds](https://github.com/photoview/photoview/discussions/127), before v2.0.0, I found that some database import files were required (they are included in the preliminary zipped builds), but I can't remember where they came from (they're not in the repo), or if they're needed with v2.0.0 or SQLite, which I'm using for testing.

**Comment by viktorstrate** [comment]
I have very little experience developing for Windows, but I'm open to pull requests, if some changes are needed to support Windows.

**Comment by stilobique** [comment]
I don't know your techno, but if i can maybe i'm try to make a fix. I'm curious to test the Go language. Thanks four your reply guys !

**Comment by DevilsDesigns** [comment]
Hey ive been trying to make a windows build for over 2 weeks. I've been trying to build with WSL2 and convert the go install to windows. I don't know your methods. But I found you can make a compile using go for windows. So it automatically converts the go build into a windows .exe by defualt. What method have you been using?

**Comment by DevilsDesigns** [comment]
> I've had the same problem, trying to run windows builds.
> 
> I suspect that something is trying to make folders with a colon in the foldername, e.g. `C:`, which is not allowed on Windows.
> 
> In my [earlier experimentation with Windows builds](https://github.com/photoview/photoview/discussions/127), before v2.0.0, I found that some database import files were required (they are included in the preliminary zipped builds), but I can't remember where they came from (they're not in the repo), or if they're needed with v2.0.0 or SQLite, which I'm using for testing.

I followed your tut instructions but to no avail. kept getting weird build errors with npm. Which is why I tried to use WSL2. Any help would.be greatly appreciated?

**Comment by zyq1569** [comment]
临时修改
在文件environment_variables.go : 中func UIPath()  对获取windows路径修改，解决 ：CreateFile \ui\C:: 问题
在文件spa.go 中func (h SpaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)  
增加 
mime.AddExtensionType(".js", "application/javascript")
 mime.AddExtensionType(".css", "text/css")
解决 页面无法打开文件
参考 https://github.com/zyq1569/photos

Bing Translator：

Temporary modifications：
In the file
 environment_variables.go:func UIPath() /  to get windows path modifications, solve the :“CreateFileuiC:: ” problem
In the file spa.go, func (h SpaHandler) ServeHTTP(w http. ResponseWriter, r *http. Request)  
increase 
mime. AddExtensionType(".js", "application/javascript")
 mime. AddExtensionType(".css", "text/css")
Fix The page cannot open the file


Refer to https://github.com/zyq1569/photos
